internal/accounts/create.go

```diff
@@ -134,3 +149,277 @@ func create(
+	log.Info("\n------------------------")
+	return onChainAccount, nil
+}
+
+func getAccountCreatedAddressWithPubKey(
+	service *services.Services,
+	pubKey crypto.PublicKey,
+	startHeight uint64,
+) (*flow.Address, error) {
+	lastHeight, err := service.Blocks.GetLatestBlockHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	flowEvents, err := service.Events.Get([]string{flow.EventAccountKeyAdded}, startHeight, lastHeight, 20, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	var address *flow.Address
+	for _, block := range flowEvents {
+		events := flowkit.NewEvents(block.Events)
+		address = events.GetAddressForKeyAdded(pubKey)
+		if address != nil {
+			break
+		}
+	}
+
+	if address == nil {
+		if lastHeight-startHeight > 200 { // if something goes wrong don't keep waiting forever to avoid spamming network
+			return nil, fmt.Errorf("failed to get the account address due to time out")
+		}
+
+		time.Sleep(time.Second * 2)
+		address, err = getAccountCreatedAddressWithPubKey(service, pubKey, startHeight)
+		if err != nil {
+			return nil, err
+		}
+
+		return address, nil
+	}
+
+	return address, nil
+}
+func saveAccount(
+	loader flowkit.ReaderWriter,
+	state *flowkit.State,
+	account *flowkit.Account,
+	network config.Network,
+) error {
+	// If using emulator, save account private key to main flow.json configuration file
+	if network == config.DefaultEmulatorNetwork() {
+		return saveAccountToMainConfigFile(state, account)
+	}
+
+	// Otherwise, save to a separate {accountName}.private.jsonfile.
+	return saveAccountToPrivateConfigFile(loader, state, account)
+}
+func saveAccountToPrivateConfigFile(
+	loader flowkit.ReaderWriter,
+	state *flowkit.State,
+	account *flowkit.Account,
+) error {
+	privateAccountFilename := fmt.Sprintf("%s.private.json", account.Name())
+
+	// Step 1: save the private version of the account (incl. the private key)
+	// to a separate JSON file.
+	err := savePrivateAccount(loader, privateAccountFilename, account)
+	if err != nil {
+		return err
+	}
+
+	// Step 2: update the main configuration file to inlcude a reference
+	// to the private account file.
+	fromFileAccount := flowkit.
+		NewAccount(account.Name()).
+		SetFromFile(privateAccountFilename)
+
+	state.Accounts().AddOrUpdate(fromFileAccount)
+
+	err = state.SaveDefault()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+func saveAccountToMainConfigFile(
+	state *flowkit.State,
+	account *flowkit.Account,
+) error {
+	state.Accounts().AddOrUpdate(account)
+	err := state.SaveDefault()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+func savePrivateAccount(
+	loader flowkit.ReaderWriter,
+	filename string,
+	account *flowkit.Account,
+) error {
+	privateState := flowkit.NewEmptyState(loader)
+	privateState.Accounts().AddOrUpdate(account)
+
+	err := privateState.Save(filename)
+	if err != nil {
+		return err
+	}
+
+	currentWd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	flowCliDir := strings.TrimSuffix(currentWd, "cmd/flow")
```

Comment:
Hmm, you shouldn't have to trim the directory like this. How are you running the CLI while developing?

This might be needed because you're running from the `cmd/flow` directory. Try running `go run ./cmd/flow/main.go` instead.

Reply:
addressed